src/ucp/api/ucp.h

```diff
@@ -1283,6 +1359,113 @@ void ucp_worker_release_address(ucp_worker_h worker, ucp_address_t *address);
 unsigned ucp_worker_progress(ucp_worker_h worker);
 
 
+/**
+ * @ingroup UCP_GROUP
+ * @brief Create a group object.
+ *
+ * This routine allocates and initializes a @ref ucp_group_h "group" object.
+ *
+ * @note The group object is allocated within context of the calling thread
+ *
+ * @param [in] worker      Worker to create a group on top of.
+ * @param [in] params      User defined @ref ucp_group_params_t configurations for the
+ *                         @ref ucp_group_h "UCP group".
+ * @param [out] group_p    A pointer to the group object allocated by the
+ *                         UCP library
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_group_create(ucp_worker_h worker,
+                              const ucp_group_params_t *params,
+                              ucp_group_h *group_p);
+
+
+/**
+ * @ingroup UCP_GROUP
+ * @brief Destroy a group object.
+ *
+ * This routine releases the resources associated with a
+ * @ref ucc_group_h "UCP group".
+ *
+ * @warning Once the UCP group destroy the group handle cannot be used with any
+ * UCP routine.
+ *
+ * The destroy process releases and shuts down all resources associated    with
+ * the @ref ucp_group_h "group".
+ *
+ * @param [in]  group       Group object to destroy.
+ */
+void ucp_group_destroy(ucp_group_h group);
```

Comment:
is this collective operation

Reply:
Please specify for each call if it is collective operations (aka, all other PEs have to invoke it)